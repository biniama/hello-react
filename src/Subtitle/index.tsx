import React, { FC, memo } from "react";
import { capitalise } from "../helpers";

type SubtitleProps = {
  text: string;
};

const Subtitle: FC<SubtitleProps> = ({ text }) => {
  return <h2>{capitalise(text)}</h2>;
};

export default memo(Subtitle);
