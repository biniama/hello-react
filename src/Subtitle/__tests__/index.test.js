import React from "react";
import snapshotRenderer from "react-test-renderer";
import Subtitle from "..";

const mockProps = [
  [{ text: "This is test text 1" }],
  [{ text: "This is test text 2" }]
];

describe("Subtitle component", () => {
  it.each(mockProps)("matches Subtitle snapshot %#", props => {
    const component = snapshotRenderer.create(<Subtitle {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
