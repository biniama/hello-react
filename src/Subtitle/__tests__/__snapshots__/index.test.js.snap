// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle component matches Subtitle snapshot 0 1`] = `
<h2>
  THIS IS TEST TEXT 1
</h2>
`;

exports[`Subtitle component matches Subtitle snapshot 1 1`] = `
<h2>
  THIS IS TEST TEXT 2
</h2>
`;
